client/client.go

```diff
@@ -282,27 +416,35 @@ func (client *Client) process(resp *Response) {
 func (client *Client) err(e error) {
 	if client.ErrorHandler != nil {
 		client.ErrorHandler(e)
+	} else {
+		client.Log(Error, e.Error()) // in case ErrorHandler is not supplied we try the Log, this might be important
 	}
 }
 
 func (client *Client) request() *request {
 	return client.requestPool.Get().(*request)
 }
 
-func (client *Client) submit(pt rt.PT, funcname string, payload []byte) (string, error) {
-	var err error
+func (client *Client) submit(pt rt.PT, funcname string, payload []byte) (handle string, err error) {
 
-	client.outbound <- client.request().submitJob(pt, funcname, IdGen.Id(), payload)
+	defer func() {
+		if e := recover(); e != nil {
+			handle = ""
+			err = e.(error)
+		}
```

Comment:
It's not really considered best practice to rely on `recover`, also this is an unsafe typecast here it _could_ error as there is no guarantee that `panic()` is called with an `error` object.